pkgs/development/python-modules/apsw/default.nix

```diff
@@ -37,23 +32,12 @@ buildPythonPackage rec {
     sed -i '/print_version_info(write)/d' tests.py
   '';
 
-  pytestFlagsArray = [
-    "tests.py"
-  ];
-
-  disabledTests = [
-    "testCursor"
-    "testLoadExtension"
-    "testShell"
-    "testVFS"
-    "testVFSWithWAL"
-    "testdb"
-  ] ++ lib.optionals stdenv.isDarwin [
-    # This is https://github.com/rogerbinns/apsw/issues/277 but
-    # because we use pytestCheckHook we need to blacklist the test
-    # manually
-    "testzzForkChecker"
-  ];
+  # Project uses custom test setup to exclude some tests by default, so using pytest
+  # requires more maintenance
+  # https://github.com/rogerbinns/apsw/issues/335
+  checkPhase = ''
+    python setup.py test
```

Comment:
I get identical results from both methods, except the way it already is is the documented way to run it upstream. So can you elaborate on why `python tests.py` should be preferred?

Reply:
I'm fine either way.
But with running tests directly we avoid the call to setup.py (which is deprecated).

Because upstream chose to run the tests in an alternative matter than usually (so pytest  can't be used directly) , we can also argue for a call to setup.py.

Like I said, I'm fine either way

Reply:
Upstream claims they are not using `setuptools` for their `test` command, which to my understanding is what's deprecated. I say "claims" because I'm not very familiar with running tests via `setuptools`, so I don't really know what qualifies as "running the tests via setuptools" and searching for docs on it I just end up finding all the information about it being deprecated. In other words is `python setup.py test` deprecated no matter what that ends up invoking? Or only if it invokes _something_ in `setuptools`?

Reply:
Maybe this is the answer: https://github.com/pytest-dev/pytest/pull/5546/files

APSW doesn't define `tests_require` in `setup.py`, doesn't create a `test=` alias in `setup.cfg, nor import from `setuptools.command.test`. So my reading of this is that upstream isn't technically using deprecated functionality, just using the same entry point.

Reply:
Yes, it seems upstream just created a function, which calls tests.py (and also calls setup())
So I guess we just leave it at `python setup.py test`?